src/Agent.Worker/ExecutionContext.cs

```diff
@@ -39,6 +39,7 @@ public interface IExecutionContext : IAgentService
         List<IAsyncCommandContext> AsyncCommands { get; }
         List<string> PrependPath { get; }
         ContainerInfo Container { get; }
+        List<ContainerInfo> SidecarContainers { get; }
```

Comment:
do we need both `Container` and `SidecarContainers`?
should we just has `List<ContainerInfo> Containers`, since you add `IsJobContainer` into `ContainerInfo`?

Reply:
I left this in because there are many locations that reference Container, as a quick way to see if the job is running in a container, for things like modifying paths, etc. It is better to do that than repeatedly interate the list of containers and see if any of them are a job container. It also more strictly enforces that there can only ever be one job container